libs/x-content/src/test/java/org/elasticsearch/xcontent/SimpleStruct.java

```diff
@@ -61,19 +50,6 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             .endObject();
     }
 
-    @Override
-    public boolean equals(Object o) {
-        if (this == o) return true;
-        if (o == null || getClass() != o.getClass()) return false;
-        SimpleStruct other = (SimpleStruct) o;
-        return i == other.i && d == other.d && Objects.equals(s, other.s);
-    }
-
-    @Override
-    public int hashCode() {
-        return Objects.hash(i, d, s);
-    }
-
     @Override
```

Comment:
Maybe possible to use the record's version of toString()?

Reply:
:+1: Good catch